apps/nova/src/components/dashboard/PastLessonsWrapper.tsx

```diff
@@ -0,0 +1,98 @@
+import { Route } from "@/types/routes";
+import { useUser } from "@litespace/headless/context/user";
+import { useInfiniteLessons } from "@litespace/headless/lessons";
+import { useFormatMessage } from "@litespace/luna/hooks/intl";
+import { PastLessonsTable } from "@litespace/luna/Lessons";
+import { Typography } from "@litespace/luna/Typography";
+import { ILesson, IUser } from "@litespace/types";
+import { useCallback, useMemo, useState } from "react";
+import BookLesson from "@/components/Lessons/BookLesson";
+import { InView } from "react-intersection-observer";
+import dayjs from "dayjs";
+
+function asLessons(list: ILesson.FindUserLessonsApiResponse["list"] | null) {
+  if (!list) return [];
+
+  return (
+    list.map((lesson) => {
+      const teacher = lesson.members.find(
+        (member) => member.role === IUser.Role.Tutor
+      );
+      return {
+        id: lesson.lesson.id,
+        start: lesson.lesson.start,
+        duration: lesson.lesson.duration,
+        tutor: {
+          id: teacher?.userId || 0,
+          name: teacher?.name || null,
+          imageUrl: teacher?.image || null,
+        },
+      };
+    }) || []
+  );
+}
+
+export const PastLessonsWrapper = () => {
+  const intl = useFormatMessage();
+  const [tutor, setTutor] = useState<number | null>(null);
+
+  const closeRebookingDialog = useCallback(() => {
+    setTutor(null);
+  }, []);
+
+  const openRebookingDialog = useCallback((tutorId: number) => {
+    setTutor(tutorId);
+  }, []);
```

Comment:
I belive the code will be more clean if we removed these two functions `closeRebookingDialog` and `openRebookingDialog` and just `setTutor` directly. WDYT?

Reply:
yea, it's cleaner. but it will be more performant if we memoized the functions, WDYT?